### PR TITLE
schema: Add new index for `customvar_flat`

### LINKED ISSUE #523: Add index on `customvar_flat.flatname`
At the moment, there is no index on `customvar_flat.flatname`:

https://github.com/Icinga/icingadb/blob/a93da02eba179305340d2fbde970a25bd3d12480/schema/mysql/schema.sql#L979-L991

There probably should be one as this is used when filtering objects by custom vars in Web.

To be discussed: should there also be one for `customvar.name`? I don't know if anything would use this at the moment.

### PR DESCRIPTION
**Pull request by nilmerg**
resolves #523

postgres changes still tbd

### CODE REVIEW

**Comment by nilmerg** [comment]
This is required, because https://github.com/Icinga/ipl-orm/pull/110 will change how Icinga DB Web filters custom variables. Without an index, the then not dependent subqueries will perform a full table scan.

I also played around with just an index on `flatname`, but that turned out to be ignored as the `flatname, flatvalue` index was still being preferred.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
New files names don't match PR milestone.

**Comment by julianbrost** [comment]
There are no new files. It's mixed up already and will be cleaned up with #602, including the schema version.

File: schema/pgsql/schema.sql [hunk 1]
```diff
@@ -1564,13 +1564,16 @@ ALTER TABLE customvar_flat ALTER COLUMN customvar_id SET STORAGE PLAIN;
 ALTER TABLE customvar_flat ALTER COLUMN flatname_checksum SET STORAGE PLAIN;
 
 CREATE INDEX idx_customvar_flat_customvar_id ON customvar_flat(customvar_id);
+CREATE INDEX idx_customvar_flat_flatname_flatvalue ON customvar_flat(flatname, flatvalue);
```
[the remark this answers is not in the document]
Comment (julianbrost) [reply schema/pgsql/schema.sql:1567]: Indeed. I tested with a 32k character long `flatvalue`, PostgreSQL was happy.

**Pull request merged**